airflow/executors/debug_executor.py

```diff
@@ -76,10 +77,14 @@ def _run_task(self, ti: TaskInstance) -> bool:
         try:
             params = self.tasks_params.pop(ti.key, {})
             ti._run_raw_task(job_id=ti.job_id, **params)  # pylint: disable=protected-access
+            ti.set_state(State.SUCCESS)
```

Comment:
Why is this needed? How was it working before this PR?

Reply:
good catch, run_raw_task does update state to SUCCESS internally, it was causing issues in previous implementation of the PR, but we have changed the design a lot since then, so maybe it's not needed anymore. let me remove it and rerun the tests to see if anything breaks.

Reply:
looks like we don't need it anymore, tests passed.